src/status/condition/mod.rs

```diff
@@ -108,6 +108,98 @@ pub struct ClusterCondition {
+            },
+            ClusterConditionType::Progressing => match self.status {
+                ClusterConditionStatus::True => "Progressing",
+                ClusterConditionStatus::False => "Not progressing",
+                ClusterConditionStatus::Unknown => "Progression unknown",
+            },
+            ClusterConditionType::ReconciliationPaused => match self.status {
+                ClusterConditionStatus::True => "Reconciling",
+                ClusterConditionStatus::False => "Not reconciling",
+                ClusterConditionStatus::Unknown => "Reconciliation unknown",
+            },
+            ClusterConditionType::Stopped => match self.status {
+                ClusterConditionStatus::True => "Stopped",
+                ClusterConditionStatus::False => "Running",
+                ClusterConditionStatus::Unknown => "Stopped status unknown",
+            },
+        };
+
+        out.fmt(f)
+    }
+}
+
+impl ClusterCondition {
+    /// Returns if the [`ClusterCondition`] is considered to be an error.
+    pub fn is_error(&self) -> bool {
+        match self.type_ {
+            ClusterConditionType::Available => match self.status {
+                ClusterConditionStatus::False | ClusterConditionStatus::Unknown => true,
+                ClusterConditionStatus::True => false,
+            },
+            ClusterConditionType::Degraded => match self.status {
+                ClusterConditionStatus::False | ClusterConditionStatus::Unknown => false,
+                ClusterConditionStatus::True => true,
+            },
+            ClusterConditionType::Progressing => match self.status {
+                ClusterConditionStatus::False | ClusterConditionStatus::Unknown => true,
+                ClusterConditionStatus::True => false,
+            },
+            ClusterConditionType::ReconciliationPaused => match self.status {
+                ClusterConditionStatus::False | ClusterConditionStatus::True => false,
+                ClusterConditionStatus::Unknown => true,
+            },
+            ClusterConditionType::Stopped => match self.status {
+                ClusterConditionStatus::True | ClusterConditionStatus::Unknown => true,
+                ClusterConditionStatus::False => false,
+            },
+        }
+    }
+
+    /// Returns a short display string. This method wraps the
+    /// [`std::fmt::Display`] implementation of the [`ClusterCondition`].
+    pub fn display_short(&self) -> String {
+        self.to_string()
+    }
+
+    /// Returns a long display string. This method uses the
+    /// [`std::fmt::Display`] implementation of the [`ClusterCondition`] and
+    /// combines it with the optional message to provide more context.
+    pub fn display_long(&self) -> String {
+        match &self.message {
+            Some(message) => format!("{}: {}", self, message),
+            None => self.to_string(),
+        }
+    }
+
+    /// Returns either a short ot long display string, This method additionally
```

Comment:
Please also update the docs to match the new solution

Reply:
Updated doc comment in 9d3bfa9